### PR TITLE
west: add monitor interface

### PR DESCRIPTION
**Pull request by sylvioalves**
This adds monitor interface using idf_monitor.py as basis.
It needs build ELF path to allow backtracing.

This PR also updates comments and code style.

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

### CODE REVIEW

**Comment by sylvioalves** [comment]
@shubhamkulkarni97 @uLipe PTAL

**Review by uLipe — APPROVED** [review]

File: west/tools.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
+# Copyright (c) 2020 Espressif Systems (Shanghai) Co., Ltd.
```
Comment (shubhamkulkarni97) [inline west/tools.py:1]:
```suggestion
# Copyright (c) 2021 Espressif Systems (Shanghai) Co., Ltd.
```

**Comment by shubhamkulkarni97** [comment]
@sylvioalves Minor suggestion

File: west/tools.py (hunk 1 above)
Comment (sylvioalves) [reply west/tools.py:1]: @shubhamkulkarni97 This file is not new. I have just renamed from `update.py` to `tool.py`. So I kept original date. Do you still think I need to update this?

Comment (shubhamkulkarni97) [reply west/tools.py:1]: LGTM

**Review by shubhamkulkarni97 — APPROVED** [review]

**Pull request merged**